open-trip-planner/src/main/scala/com/azavea/busplan/routing/StudentToStopRouting.scala

```diff
@@ -8,22 +8,33 @@ import scala.collection.JavaConverters._
 
 object StudentToStopRouting {
 
+  /**
+   * Find all stops that a student is allowed to given a maximum walk
+   * distance threshold. Write these results to a csv.
+   *
+   * @param path            File path for output csv
+   * @param results         Map of student IDs to stop ID-distance pairs
+   * @param getMaxDistance  Function taking a grade level and returning a distance
+   * @param studentToInfo   Map of student IDs and grade-existing stop ID pairs
+   */
   def createStudentToStopCSV(
     path: String,
     results: Map[String, List[(String, Double)]],
-    maxDistance: Double,
+    getMaxDistance: Int => Double,
     studentToInfo: Map[String, (Int, String)]): Unit = {
-    val csv = new FileWriter(path, true)
+    val csv = new FileWriter(path, false)
     val bw = new BufferedWriter(csv)
     for ((k, v) <- results) {
       bw.write(k)
-      if (v(0)._2 >= 7920) {
+      var eligibleStops = getStopsBelowThreshold(v,
+        getMaxDistance(studentToInfo(k)._1))
+      // If a student's existing stop is more than 1.5 miles from 
+      // her home, her only option is that stop
+      if (v(0)._2 >= Constants.ONE_AND_A_HALF_MILES || eligibleStops.length == 0) {
```

Comment:
This ensures that the it will always append the student's existing stop to the eligibility matrix if it doesn't find any others below the threshold. This doesn't change anything about the routes that we got from the solver because the solver already does this but I figured that it may be cleaner to also add it to this step.